tests/dummy/app/templates/components/validated-input.hbs

```diff
@@ -5,10 +5,16 @@
   {{/if}}
 
   <div class="input-error">
-      {{#if showMessage}}
+      {{#if showErrorMessage}}
           <div class="error">
-              {{v-get model valuePath 'message'}}
+              {{get (v-get model valuePath 'error') 'message'}}
           </div>
       {{/if}}
+
+      {{#if showWarningMessage}}
+        <div class="warning">
+            {{get (v-get model valuePath 'warning') 'message'}}
+        </div>
+      {{/if}}
```

Comment:
Ah, of course - I blanked that messages are strings on the current API. Definitely don't want to break that.

Though warnings are optional, it seems like anytime someone took the time to write a validation for a warning, they'd want to expose it to the user, likely just with another class to style differently.